pandas/tests/groupby/test_groupby.py

```diff
@@ -1463,13 +1495,23 @@ def f(group):
     assert names == expected_names
 
 
-def test_no_dummy_key_names(df):
+def test_no_dummy_key_names(df, using_infer_string):
     # see gh-1291
-    result = df.groupby(df["A"].values).sum()
```

Comment:
The same comment here (convert to object dtype to preserve the actual test)

Alternatively, we should maybe have a `df` with only one string column, so this `df.groupby("A").sum()` cases just keep working